File: Back/.Net/Modules/Bets/BetFriends.Bets.Application.UnitTests/Implems/CompleteBetPresenter.cs
using BetFriends.Bets.Application.Features.CompleteBet;

namespace BetFriends.Bets.Application.UnitTests.Implems;

internal class CompleteBetPresenter : ICompleteBetOutputPort
{
    public string ErrorMessage { get; private set; } = null!;
    internal CompleteBetResponse Response { get; private set; }

    public void BetDoesNotExist(Guid betId)
    {
        ErrorMessage = $"bet {betId} does not exist";
    }

    public void Success(CompleteBetResponse completeBetResponse)
    {
        Response = completeBetResponse;
    }
}

File: Back/NodeJS/api/nest/src/bet/features/complete-bet/CompleteBetPresenter.ts
import { Presenter } from "src/Presenter";
import { FastifyReply } from 'fastify';
import { HttpStatus } from '@nestjs/common';
import { CompleteBetResponse, ICompleteBetOutputPort } from "../../../../../../modules/bets/src/application/features/complete-bet/CompleteBetHandler";

export class CompleteBetPresenter extends Presenter implements ICompleteBetOutputPort {
    BuildResponse(res: FastifyReply) {
        res.code(this.response.code).send(this.response.body)
    }
    
    UnAuthorized(userId: string): void {
        this.response = { code: HttpStatus.BAD_REQUEST, body: { Message: `User ${userId} is not authorize to close this bet` } }
    }
    SuccessfulBetNeedsProof(): void {
        this.response = { code: HttpStatus.BAD_REQUEST, body: { Message: `A successful bet needs a proof` } }
    }
    BetDoesNotExist(betId: string): void {
        this.response = { code: HttpStatus.BAD_REQUEST, body: { Message: `Bet ${betId} does not exist` } }
    }
    Present(response: CompleteBetResponse): void {
        this.response = { code: HttpStatus.OK, body: {  } }
    }

}